src/world/World.php

```diff
@@ -2951,7 +2951,7 @@ function() use ($resolver) : void{
 	 * @throws WorldException if the terrain is not generated
 	 */
 	public function getSafeSpawn(?Vector3 $spawn = null) : Position{
-		if(!($spawn instanceof Vector3) || $spawn->y < 1){
+		if($spawn === null || $spawn->y < $this->minY){
```

Comment:
```suggestion
		if($spawn === null || $spawn->y < $this->minY + 1){
```
There must be at least one block where the player can stand otherwise, the player will spawn above the void.

Reply:
<= can be used instead